Framework/PythonInterface/mantid/_testing/AssertAlmostEqual.py

```diff
@@ -26,23 +27,26 @@ def assert_almost_equal(Workspace1, Workspace2, rtol=0.0, atol=0.0):
     AssertionError
         If Workspace1 and Workspace2 are not equal up to specified precision
     ValueError
-        If atol and rtol are both provided
+        If atol and rtol are both not provided
 
     """
-
-    if rtol != 0.0 and atol != 0.0:
-        raise ValueError("Specify rtol or atol, not both")
-    _rel = False
-    tolerance = 1e-10
-    if atol:
-        tolerance = atol
-
-    if rtol:
-        tolerance = rtol
-        _rel = True
-
-    result, message = CompareWorkspaces(Workspace1, Workspace2, Tolerance=tolerance, ToleranceRelErr=_rel)
-    msg_dict = message.toDict()
-    if not result:
-        msg = ", ".join(msg_dict["Message"]) + f", Workspaces are not within tolerance ({tolerance})"
-        raise AssertionError(msg)
+    # check arguments
+    if len(set(kwargs.keys()).intersection({"Workspace1", "Workspace2", "Tolerance", "ToleranceRelErr"})):
+        raise ValueError("Workspace1, Workspace2, Tolerance, ToleranceRelErr cannot be passed as additional parameters")
+
+    if rtol == Property.EMPTY_DBL and atol == Property.EMPTY_DBL:
+        raise ValueError("Specify rtol or atol")
+
+    if rtol > Property.EMPTY_DBL:
```

Comment:
`Property.EMPTY_DBL` is [equivalent to `DBL_MAX` in C++](https://github.com/mantidproject/mantid/blob/65408482e579ec532b373a45032b5ca40a45ec52/Framework/Kernel/inc/MantidKernel/EmptyValues.h#L43).  This condition will _never_ be true.  Likewise for `atol` below.  The result is that this test fixture will _always_ pass, even if the workspaces are not equal.

Reply:
`Property.EMPTY_DBL` is a special value that optional float properties can be set to. For interaction with `CompareWorkspaces`, both `atol` and `rtol` are (separately) passed in as [Tolerance parameter](https://github.com/mantidproject/mantid/blob/main/Framework/Algorithms/src/CompareWorkspaces.cpp#L123) and read by the algorithm [here](https://github.com/mantidproject/mantid/blob/main/Framework/Algorithms/src/CompareWorkspaces.cpp#L123). This line should probably be changed to verify that the value isn't set to `EMPTY_DBL`.